prova

calvinmetcalf opened the pull request:
hopefully testing on firefox in travis

calvinmetcalf commented:
this is tricky because I'm pretty sure firefox is getting the script not responding box which is impossible to click

calvinmetcalf commented:
ok got it working by asyncifying the pbkdf2 and omitting much of the diffie-hellman,

dcousens commented on test/dh.js:
```diff
@@ -22,26 +22,28 @@ test('diffie-hellman mod groups', function (t) {
-      var pub1 = dh1.computeSecret(pubk2).toString('hex');
-      var pub2 = dh2.computeSecret(dh1.getPublicKey()).toString('hex');
-      t.equals(pub1, pub2, 'equal secrets');
+if (!process.browser) {
```
can you elaborate on this?  I'm not sure I follow.

calvinmetcalf replied:
these tests take a LONG time so on a very very slow travis browser they time out/lock the browser

dcousens replied:
I feel like this is just a travis artifact, and shouldn't be omitted from general browser testing.
How would I test this code in a browser manually without having to omit these statements?

dcousens commented on test/pbkdf2.js:
```diff
@@ -1,23 +1,24 @@
   vectors.forEach(function (input) {
-    //skip inputs that will take way too long
-    if(input.iterations > 10000) return
+  	setTimeout(function () {
```
Why?  I know you mentioned it, but this seems like a hack.

calvinmetcalf replied:
its to prevent the script from locking the page up to long,

calvinmetcalf commented:
ok I removed the prepublish script and replaced it with one called publish which runs npm ls, followed by npm run test-node and then runs npm publish, this is because republish runs after npm install but if the tests fail there then instead of just failing travis it causes npm install to be fun again up to 3 times.

This changes the tests in 2 ways

- the pbkdf2 tests are changed to be async, this is to prevent locking the browser which ( I believe) causes a script not responding dialog to show up in firefox which we can't then click through so the test hangs
- the diffie-hellman tests are truncated to only do a few of the comparisons and not generate any primes.   Again this is do to them taking a while and locking up the browser, I might be able to do something overly complicated with a web worker but that seems like a phase 2 to me.

Before merging I can squash

dominictarr commented:
:+1:

calvinmetcalf commented:
ok rebased and squashed going to make sure it's still green

dcousens commented:
Before merge, can we ensure we aren't just disabling tests simply because travis times out?

calvinmetcalf commented:
Ok won't merge yet

On Mon, Jan 19, 2015, 7:10 PM Daniel Cousens <notifications@github.com>
wrote:

> Before merge, can we ensure we aren't just disabling tests simply because
> travis times out?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crypto-browserify/crypto-browserify/pull/111#issuecomment-70583338>
> .
>

calvinmetcalf commented:
trying to get the diffie-hellmen tests to work in a web worker leading to some circuler deps issues, looks like diffie-hellmen will be the next one to break out

calvinmetcalf commented:
closing as it's bit rotted
